playbooks/weekly-scan.yaml

```diff
@@ -0,0 +1,151 @@
+---
+- name: Run pentesting tools
+  hosts: all
+  tasks:
+    - name: Run naabu scan
```

Comment:
Why do you add the same tests into daily and weekly scans if the daily ones are going to be executed anyway? Even worse is that they will start at the same time (daily and weekly). From the composability and reusability (and parallelization) point of view it makes much more sense to split different tests (that do not depend on each other) into independent jobs that are included as necessary into the pipelines.

Reply:
The only same "tests" are the scans performed by ProjectDiscovery tools (naabu-httpx-nuclei). Those are mandatory as the ouput of each tool is chained into the next one and, in the case of ZAP, we need to input URL (protocol://fqdn:port) targets to the scanner and not plain IPs.

As already discussed in the aligned [Pentesting Methodology](https://scs.sovereignit.de/nextcloud/apps/files/?dir=/&fileid=228392), targets for the autmated infra scanning are IP addresses. Naabu scans for open ports on them and httpx identifies web services that are used as inputs for other tools in the pipeline.

Regarding other tools used, in daily scans ZAP is launched in passive mode, while in weekly scans it is launched in full active mode, hence providing more detailed results.

The results of both daily and weekly scans are then sent to DefectDojo at different "engagements". Each engagement provides a comprehensive and detailed view of the vulnerabilities found, removing automatically those that have been solved or mitigated (as an example). Due to this, if we remove things as they are from the pipelines we will loose that capability of tracking vulnerabilities.

Reply:
After long time looking at the steps I see you share the date using the volumes. Well, while this is not bad on it's own it is absolutely unclear for a person looking at the playbooks. You should definitely comment your code.
Anyway, what you could do to improve maintainability and reusability is to split things into individual playbooks and reuse them (i.e. like https://opendev.org/opendev/base-jobs/src/branch/master/playbooks/infra-prod/pre.yaml). Something like playbook "naabu" with 2 steps (starting container and waiting for it to complete), "httpx" one, "nuclei", etc. Keep different tests (even it they are chained) independently and reuse as much as possible. This will reduce code duplication and improve usability. This are the basics rules of playbooks and especially CI jobs creation.

Reply:
I take your point and it makes sense. Will take care of it and create separate playbooks for each step.

Code commenting and documentation creation is a WIP right now.

Reply:
Don't over-optimize it :wink:  
We defined the pipeline parts in the original concept (six in total, including DefectDojo) and they are covered in Ansible now. 

* Yes, please add documentation where beneficial. Priority is filling the markdown files in `docs/` to be rendered on docs.scs.community
* Documentation of Ansible playbooks can follow, after merge of this PR
* Splitting Ansible roles and modules is not needed at the moment